pkg/web/api/v1/account.go

```diff
@@ -42,8 +42,9 @@ type CryptoAddress struct {
 }
 
 type AccountsList struct {
-	Page     *PageQuery `json:"page"`
-	Accounts []*Account `json:"accounts"`
+	Page          *PageQuery `json:"page"`
+	Accounts      []*Account `json:"accounts"`
+	HasPermission bool       `json:"permission"`
```

Comment:
Amending the response seemed like the best way to check a user's role or if they have permission to perform certain actions in the partial files.

Reply:
I do get the motivation to do this - particularly as the route handler is the thing that does the permission checking. The problem is that amending this response will also return that bool to API users and permissions in API responses isn't best practice or at the very least would require extensive documentation. 

I think we need to delegate all permissions operations to the `scene`, and that might mean we have to update all of our content negotiation as follows:

```golang
	c.Negotiate(http.StatusOK, gin.Negotiate{
		Offered:  []string{binding.MIMEJSON, binding.MIMEHTML},
		Data:     out,
		HTMLData: scene.New().WithContext(out).WithPermission(foo),
		HTMLName: "account_list.html",
	})
```

Where `WithContext` allows the scene to capture the API context and then we add our permission handling to the scene instead of the api response. What do you think about this approach?

Reply:
Yeah, that was my concern. When I tried sending the context via `HTMLData` none of the data appeared and I wasn't sure if we wanted to do it this way or add the data to the context, though I didn't think of changing the content negotiation in this way. I'm fine with adopting this approach.